File: 2023/day1/part1.py
total = 0

with open("./input.txt", "r") as f:
    for line in f.readlines():
        first = ""
        last = ""
        for char in line:
            try:
                n = str(int(char))
                if first == "":
                    first = n
                last = n
            except:
                pass
        number = int(first + last)
        total += number

print(total)

File: 2021/day4/part1.ts
import * as fs from "fs/promises"
import { Board } from "./Board"

const text = await fs.readFile("./input.txt", "utf-8")
const [numbersText, ...boardsText] = text.split("\n\n")
const boards = boardsText
  .map((board) =>
    board.split("\n").map((line) =>
      line
        .split(/\s+/)
        .map((n) => Number.parseInt(n))
        .filter((n) => !Number.isNaN(n))
    )
  )
  .map((b) => new Board(b))

const numbers = numbersText.split(",").map((n) => Number(n))
let lastNumber: number = 0
let currentNumber: number = numbers.shift() ?? -1
if (currentNumber === -1) throw new Error("No numbers ??")

while (!boards.some((board) => board.won)) {
  boards.forEach((board) => board.select(currentNumber))
  lastNumber = currentNumber
  currentNumber = numbers.shift() ?? -1
  if (currentNumber === -1) throw new Error("No numbers ??")
}

const winningBoard = boards.find((board) => board.won)
console.log(winningBoard)
console.log(winningBoard?.score)
